src/main/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountExtractServiceImpl.java

```diff
@@ -0,0 +1,765 @@
+            // ACH detail is not for a checking or savings account.
+            appendFailurePrefix(failureMessage, achDetail.getNetID(), listIndex++);
+            failureMessage.append(" Account is not a checking or savings account. ");
+        }
+        
+        // Log the message as a warning if non-blank, and also treat non-blank messages as failures; otherwise return success.
+        if (failureMessage.length() > 0) {
+            LOG.warn(failureMessage.toString());
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Helper method invoked by validateACHBatchDetail() when an ACH batch detail is not
+     * valid for processing and the failure message needs to have a prefix added for the
+     * first or subsequent failures detected.
+     * 
+     * @param failureMessage The StringBuilder to append the prefix to.
+     * @param netID The payee's netID.
+     * @param listIndex The index of the next failure reason to be appended by the calling code.
+     */
+    protected void appendFailurePrefix(StringBuilder failureMessage, String netID, int listIndex) {
+        if (failureMessage.length() == 0) {
+            failureMessage.append("ACH Detail for payee \"").append(netID).append("\" could not be processed for the following reasons: ");
+        }
+        failureMessage.append('[').append(listIndex).append(']');
+    }
+
+
+
+    /**
+     * Creates and routes a PAAT document to create a new ACH Account of the given payee type.
+     */
+    protected String addACHAccount(Person payee, PayeeACHAccountExtractDetail achDetail, String payeeType) {
+        // Create and route a new PAAT (Payee ACH Account Maintenance) document.
+        try {
+            // Create document and set description.
+            MaintenanceDocument paatDocument = (MaintenanceDocument) documentService.getNewDocument(CUPdpConstants.PAYEE_ACH_ACCOUNT_EXTRACT_MAINT_DOC_TYPE);
+            paatDocument.getDocumentHeader().setDocumentDescription(getDocumentDescription(payee, PayeeIdTypeCodes.ENTITY.equals(payeeType), true));
+            
+            // Configure as "New" maintenance and get maintained object.
+            PayeeACHAccountMaintainableImpl maintainable = (PayeeACHAccountMaintainableImpl) paatDocument.getNewMaintainableObject();
+            maintainable.setMaintenanceAction(KFSConstants.MAINTENANCE_NEW_ACTION);
+            PayeeACHAccount achAccount = (PayeeACHAccount) maintainable.getDataObject();
+            
+            // Setup payee ID and type.
+            if (PayeeIdTypeCodes.ENTITY.equals(payeeType)) {
+                achAccount.setPayeeIdNumber(payee.getEntityId());
+            } else if (PayeeIdTypeCodes.EMPLOYEE.equals(payeeType)) {
+                achAccount.setPayeeIdNumber(payee.getEmployeeId());
+            } else {
+                throw new WorkflowException("Unexpected payee ID type for automated Payee ACH Account creation: " + payeeType);
+            }

```

Comment:
I believe that this should also be changed to returning a boolean and logging the info about the problem instead of throwing an exception.  Daniela and ChadH : your thoughts?